agent/src/apiclient.rs

```diff
@@ -232,12 +233,15 @@ pub(super) mod api {
     /// Extract error statuscode from stderr string
     /// Error Example:
     /// "Failed POST request to '/actions/refresh-updates': Status 423 when POSTing /actions/refresh-updates: Update lock held\n"
-    fn extract_status_code_from_error(error: &str) -> &str {
-        let error_content_split_by_status: Vec<&str> = error.split("Status").collect();
-        let error_content_split_by_whitespace: Vec<&str> = error_content_split_by_status[1]
+    fn extract_status_code_from_error(error: &str) -> Option<StatusCode> {
+        error
+            .split("Status ")
+            .nth(1)?
```

Comment:
We should create a error enum variant like StatusCodeParsingError in https://github.com/bottlerocket-os/bottlerocket-update-operator/blob/c0d2f62a781b5a60e2af8c497ba9eed46cb09e3e/agent/src/agentclient.rs#L679 and propagate that error with .context(). You can refer to examples within this file. Same thing for next()? below. We want to be intentional about the error we throw when bail early.